packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItem.tsx

```diff
@@ -181,6 +142,7 @@ export const TreeItem = ({
           showContextMenu: true,
           menuAnchor: currentTarget,
           editingMode: false,
+          editingkey: '',
```

Comment:
Could you instead use prevState to stop calculating everything from scratch here and in the next two examples of this PR